url270fc4f5ee_ykcilborw_Joust_tgz-pJ8-fri_patterns_interpreter_parsergenerator_examples_HowToOverrideLexerImplJ8/src/com/wroblicky/andrew/joust/game/GameSetup.java

```diff
@@ -0,0 +1,505 @@
+		actives.add(bp5);
+		actives.add(bp6);
+		actives.add(bp7);
+		actives.add(bp8);
+		
+		// white castles
+		actives.add(wc1);
+		actives.add(wc2);
+		
+		// black castles
+		actives.add(bc1);
+		actives.add(bc2);
+		
+		// white knights
+		actives.add(wh1);
+		actives.add(wh2);
+		
+		// black knights
+		actives.add(bh1);
+		actives.add(bh2);
+		
+		// white bishops
+		actives.add(wb1);
+		actives.add(wb2);
+		
+		// black bishops
+		actives.add(bb1);
+		actives.add(bb2);
+		
+		// queens
+		actives.add(bq1);
+		actives.add(wq1);
+		
+		// kings
+		actives.add(bk1);
+		actives.add(wk1);
+		
+		// white pawns
+		chessBoard.addChessPiece(wp1, chessBoard.getLocation("a2"));
+		chessBoard.addChessPiece(wp2, chessBoard.getLocation("b2"));
+		chessBoard.addChessPiece(wp3, chessBoard.getLocation("c2"));
+		chessBoard.addChessPiece(wp4, chessBoard.getLocation("d2"));
+		chessBoard.addChessPiece(wp5, chessBoard.getLocation("e2"));
+		chessBoard.addChessPiece(wp6, chessBoard.getLocation("f2"));
+		chessBoard.addChessPiece(wp7, chessBoard.getLocation("g2"));
+		chessBoard.addChessPiece(wp8, chessBoard.getLocation("h2"));
+		
+		// white royals
+		chessBoard.addChessPiece(wc1, chessBoard.getLocation("a1"));
+		chessBoard.addChessPiece(wh1, chessBoard.getLocation("b1"));
+		chessBoard.addChessPiece(wb1, chessBoard.getLocation("c1"));
+		chessBoard.addChessPiece(wq1, chessBoard.getLocation("d1"));
+		chessBoard.addChessPiece(wk1, chessBoard.getLocation("e1"));
+		chessBoard.addChessPiece(wb2, chessBoard.getLocation("f1"));
+		chessBoard.addChessPiece(wh2, chessBoard.getLocation("g1"));
+		chessBoard.addChessPiece(wc2, chessBoard.getLocation("h1"));
+		
+		// black pawns
+		chessBoard.addChessPiece(bp1, chessBoard.getLocation("a7"));
+		chessBoard.addChessPiece(bp2, chessBoard.getLocation("b7"));
+		chessBoard.addChessPiece(bp3, chessBoard.getLocation("c7"));
+		chessBoard.addChessPiece(bp4, chessBoard.getLocation("d7"));
+		chessBoard.addChessPiece(bp5, chessBoard.getLocation("e7"));
+		chessBoard.addChessPiece(bp6, chessBoard.getLocation("f7"));
+		chessBoard.addChessPiece(bp7, chessBoard.getLocation("g7"));
+		chessBoard.addChessPiece(bp8, chessBoard.getLocation("h7"));
+		
+		// black royals
+		chessBoard.addChessPiece(bc1, chessBoard.getLocation("a8"));
+		chessBoard.addChessPiece(bh1, chessBoard.getLocation("b8"));
+		chessBoard.addChessPiece(bb1, chessBoard.getLocation("c8"));
+		chessBoard.addChessPiece(bq1, chessBoard.getLocation("d8"));
+		chessBoard.addChessPiece(bk1, chessBoard.getLocation("e8"));
+		chessBoard.addChessPiece(bb2, chessBoard.getLocation("f8"));
+		chessBoard.addChessPiece(bh2, chessBoard.getLocation("g8"));
+		chessBoard.addChessPiece(bc2, chessBoard.getLocation("h8"));
+	
+		Game game = new Game(chessBoard, new ArrayList<Turn>(), new ChessPieceSubsetManager(actives));
+		GameManager gameManager = new GameManager(game);
+		
+		return gameManager;
+	}
+
+	public static GameManager setupSpecialLayout(String initialConfig) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 124 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 338-503 into a separate method.